src/main/java/org/springframework/integration/aws/support/S3SessionFactory.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.aws.support;
+
+import org.springframework.integration.file.remote.session.SessionFactory;
+import org.springframework.integration.file.remote.session.SharedSessionCapable;
+import org.springframework.util.Assert;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+
+/**
+ * An Amazon S3 specific {@link SessionFactory} implementation.
+ * Also this class implements {@link SharedSessionCapable} around the single instance,
+ * since the {@link S3Session} is simple thread-safe wrapper for the {@link AmazonS3}.
+ *
+ * @author Artem Bilan
+ */
+public class S3SessionFactory implements SessionFactory<S3ObjectSummary>, SharedSessionCapable {
```

Comment:
Yeah... If just only for consistency...
In fact there is no any other S3 `SessionFactory` impl, so that `Default` prefix does not make so much sense. Otherwise someone may assume anything else, but he won't find.

Finally that may end up with the just as a matter of opinion, but I'd prefer to leave the name like this, even if we have that extra prefix in other legacy places.